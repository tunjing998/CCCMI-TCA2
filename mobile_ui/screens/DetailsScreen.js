import React, {useEffect, useState} from 'react';
import {View, StyleSheet, Text, Button, FlatList} from 'react-native';
import {ListItem} from 'react-native-elements';
import {useTheme} from '@react-navigation/native';
import testVariables from '../appium_automation_testing/test_variables';
import AsyncStorage from '@react-native-community/async-storage';

const DetailsScreen = ({navigation, route}) => {
  const {colors} = useTheme();
  const theme = useTheme();
  const [river, setRiver] = useState();

  const styles = StyleSheet.create({
    container: {
      flex: 1,
    },
    title: {
      color: colors.text,
    },
    listContainer: {
      backgroundColor: colors.background,
    },
  });

  const getData = async () => {
    try {
      const jsonValue = await AsyncStorage.getItem('river');
      return jsonValue != null ? setRiver(JSON.parse(jsonValue)) : null;
    } catch (e) {
      // error reading value
    }
  };

  useEffect(() => {
    getData();
  }, []);

  const renderDetails = () => {
    return (
      <View
        style={styles.listContainer}
        accessibilityLabel={testVariables.riverDetailScreen}
        testID={testVariables.riverDetailScreen}>
        <ListItem bottomDivider containerStyle={styles.listContainer}>
          <ListItem.Content>
            <ListItem.Subtitle style={styles.title}>
              River Name
            </ListItem.Subtitle>
            <ListItem.Title style={styles.title}>
              {river.river_name}
            </ListItem.Title>
          </ListItem.Content>
        </ListItem>
        <ListItem bottomDivider containerStyle={styles.listContainer}>
          <ListItem.Content>
            <ListItem.Subtitle style={styles.title}>Latitude</ListItem.Subtitle>
            <ListItem.Title style={styles.title}>
              {river.latitude}
            </ListItem.Title>
          </ListItem.Content>
        </ListItem>
        <ListItem bottomDivider containerStyle={styles.listContainer}>
          <ListItem.Content>
            <ListItem.Subtitle style={styles.title}>
              Longitude
            </ListItem.Subtitle>
            <ListItem.Title style={styles.title}>
              {river.longitude}
            </ListItem.Title>
          </ListItem.Content>
        </ListItem>
        <ListItem bottomDivider containerStyle={styles.listContainer}>
          <ListItem.Content>
            <ListItem.Subtitle style={styles.title}>
              River Catchments
            </ListItem.Subtitle>
            <ListItem.Title style={styles.title}>
              {river.river_catchments}
            </ListItem.Title>
          </ListItem.Content>
        </ListItem>
        <ListItem bottomDivider containerStyle={styles.listContainer}>
          <ListItem.Content>
            <ListItem.Subtitle style={styles.title}>
              River Code
            </ListItem.Subtitle>
            <ListItem.Title style={styles.title}>
              {river.river_code}
            </ListItem.Title>
          </ListItem.Content>
        </ListItem>
        <ListItem bottomDivider containerStyle={styles.listContainer}>
          <ListItem.Content>
            <ListItem.Subtitle style={styles.title}>
              Local Authority
            </ListItem.Subtitle>
            <ListItem.Title style={styles.title}>
              {river.local_authority}
            </ListItem.Title>
          </ListItem.Content>
        </ListItem>

        <ListItem bottomDivider containerStyle={styles.listContainer}>
          <ListItem.Content>
            <ListItem.Subtitle style={styles.title}>
              Transboundary
            </ListItem.Subtitle>
            <ListItem.Title style={styles.title}>
              {river.transboundary}
            </ListItem.Title>
          </ListItem.Content>
        </ListItem>
      </View>
    );
  }

  return (
    <View>{river && renderDetails()}</View>
  )
  
};

export default DetailsScreen;
