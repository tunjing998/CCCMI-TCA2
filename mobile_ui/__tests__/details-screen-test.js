import React, {Component} from 'react';
import {View, Text} from 'react-native';
import DetailsScreen from '../screens/DetailsScreen';
import renderer from 'react-test-renderer';

test('DetailsScreen snapShot', () => {
  const snap = renderer.create(<DetailsScreen />).toJSON();
  expect(snap).toMatchSnapshot();
});
