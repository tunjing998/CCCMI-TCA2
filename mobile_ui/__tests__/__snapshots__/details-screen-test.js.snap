// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsScreen snapShot 1`] = `
<View
  accessibilityLabel="riverDetailScreen"
  style={
    Object {
      "backgroundColor": "rgb(242, 242, 242)",
    }
  }
  testID="riverDetailScreen"
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(242, 242, 242)",
          "borderBottomWidth": 0.5,
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="padView"
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "center",
          }
        }
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgb(28, 28, 30)",
              "fontSize": 15,
            }
          }
          testID="listItemTitle"
        >
          River Name
        </Text>
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgb(28, 28, 30)",
              "fontSize": 17,
            }
          }
          testID="listItemTitle"
        >
          River Liffey
        </Text>
      </View>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(242, 242, 242)",
          "borderBottomWidth": 0.5,
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="padView"
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "center",
          }
        }
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgb(28, 28, 30)",
              "fontSize": 15,
            }
          }
          testID="listItemTitle"
        >
          Latitude
        </Text>
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgb(28, 28, 30)",
              "fontSize": 17,
            }
          }
          testID="listItemTitle"
        >
          53.3492319
        </Text>
      </View>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(242, 242, 242)",
          "borderBottomWidth": 0.5,
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="padView"
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "center",
          }
        }
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgb(28, 28, 30)",
              "fontSize": 15,
            }
          }
          testID="listItemTitle"
        >
          Longitude
        </Text>
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgb(28, 28, 30)",
              "fontSize": 17,
            }
          }
          testID="listItemTitle"
        >
          -6.535848
        </Text>
      </View>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(242, 242, 242)",
          "borderBottomWidth": 0.5,
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="padView"
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "center",
          }
        }
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgb(28, 28, 30)",
              "fontSize": 15,
            }
          }
          testID="listItemTitle"
        >
          Area (km²)
        </Text>
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgb(28, 28, 30)",
              "fontSize": 17,
            }
          }
          testID="listItemTitle"
        >
          23.58
        </Text>
      </View>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(242, 242, 242)",
          "borderBottomWidth": 0.5,
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="padView"
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "center",
          }
        }
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgb(28, 28, 30)",
              "fontSize": 15,
            }
          }
          testID="listItemTitle"
        >
          Length (km)
        </Text>
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgb(28, 28, 30)",
              "fontSize": 17,
            }
          }
          testID="listItemTitle"
        >
          14.51
        </Text>
      </View>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(242, 242, 242)",
          "borderBottomWidth": 0.5,
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="padView"
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "center",
          }
        }
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgb(28, 28, 30)",
              "fontSize": 15,
            }
          }
          testID="listItemTitle"
        >
          Ecological Status
        </Text>
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgb(28, 28, 30)",
              "fontSize": 17,
            }
          }
          testID="listItemTitle"
        >
          Good
        </Text>
      </View>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(242, 242, 242)",
          "borderBottomWidth": 0.5,
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="padView"
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "center",
          }
        }
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgb(28, 28, 30)",
              "fontSize": 15,
            }
          }
          testID="listItemTitle"
        >
          Local Authority
        </Text>
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgb(28, 28, 30)",
              "fontSize": 17,
            }
          }
          testID="listItemTitle"
        >
          Kildare County Council
        </Text>
      </View>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(242, 242, 242)",
          "borderBottomWidth": 0.5,
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="padView"
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "center",
          }
        }
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgb(28, 28, 30)",
              "fontSize": 15,
            }
          }
          testID="listItemTitle"
        >
          Protected Area
        </Text>
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgb(28, 28, 30)",
              "fontSize": 17,
            }
          }
          testID="listItemTitle"
        >
          Yes
        </Text>
      </View>
    </View>
  </View>
</View>
`;
